### DISCUSSION TITLE
TQDM progress bar in Github Action terminal printing each case in a separate line

### DISCUSSION THREAD

**Discussion by Zanger67** [tqdm/tqdm#1600] [Q&A]
Not sure if this is just due to the limitations of GitHub Action's terminal output (e.g. its response to carriage returns) but using TQDM in it I've noticed it will reprint the TQDM statement every few increments. 

Notably, it won't do it on each `.update(1)` but every few (most of the time it's every ~1% difference but it also occasionally repeats the same percentage with a higher number).

GitHub action sample image:
![image](https://github.com/user-attachments/assets/23aef15d-b487-4ed8-b738-e1ac0a17865d)

In local instances, both with in jupyter notebooks (with the `tqdm.notebook` version) and in regular scripts, it works fine so I don't think it's an issue with implementation.

Regular case in powershell:
![image](https://github.com/user-attachments/assets/ac7e089c-3789-4509-9e4f-8cb0fc5d0ec3)

Anyone have any insight into this and/or if it's possible to have the TQDM statement work per usual in GitHub actions? Or if someone knows if it's due to GitHub actions and there's no avoiding it. :l

The code I have implementing TQDM (the commented out line was another I tried that yielded the same result): 
```python
    with tqdm(total=len(paths)) as pbar :
    # with tqdm(total=len(paths), position=0, leave=True) as pbar :
        for i, path in enumerate(paths) :
            path = join(LCPATH, path)
            output[path] = individualCTimeViaGit(cmd + [path])
            pbar.update(1)
```

Import statement (confirmed with print statements that the correct one is being imported)
```python
if IS_NOTEBOOK :
    print('Importing tqdm.notebook')
    from tqdm.notebook import tqdm
else :
    print('Importing tqdm (non-notebook)')
    from tqdm import tqdm
```

**Comment by evguenipetkov** [comment]
I think it is an issue in gihub.
